be/src/storage/update_manager.cpp

```diff
@@ -283,6 +283,46 @@ void UpdateManager::clear_cached_delta_column_group(const std::vector<TabletSegm
     }
 }
 
+Status UpdateManager::set_cached_empty_delta_column_group(KVStore* meta, const TabletSegmentId& tsid) {
+    {
+        std::lock_guard<std::mutex> lg(_delta_column_group_cache_lock);
+        auto itr = _delta_column_group_cache.find(tsid);
+        if (itr != _delta_column_group_cache.end()) {
+            // already exist, not need to cache
+            return Status::OK();
+        }
+    }
+    // find from rocksdb
+    DeltaColumnGroupList new_dcgs;
+    RETURN_IF_ERROR(
+            TabletMetaManager::get_delta_column_group(meta, tsid.tablet_id, tsid.segment_id, INT64_MAX, &new_dcgs));
+    std::lock_guard<std::mutex> lg(_delta_column_group_cache_lock);
+    auto itr = _delta_column_group_cache.find(tsid);
+    if (itr != _delta_column_group_cache.end()) {
+        // already exist, not need to cache
+        return Status::OK();
+    }
+    if (new_dcgs.empty()) {
+        // only set empty dcgs
+        _delta_column_group_cache[tsid] = new_dcgs;
+    }
+    return Status::OK();
+}
+
+bool UpdateManager::TEST_get_cached_delta_column_group(const TabletSegmentId& tsid, int64_t version,
+                                                       DeltaColumnGroupList* dcgs) {
+    // find in delta column group cache
+    std::lock_guard<std::mutex> lg(_delta_column_group_cache_lock);
+    auto itr = _delta_column_group_cache.find(tsid);
+    if (itr != _delta_column_group_cache.end()) {
+        search_delta_column_groups_by_version(itr->second, version, dcgs);
```

Comment:
Do you need to check the status of `search_delta_column_groups_by_version`?

Reply:
Not need that, `search_delta_column_groups_by_version` return void